cfme/configure/tasks.py

```diff
@@ -105,6 +109,167 @@ def is_analysis_finished(name, task_type='vm', clear_tasks_after_success=True):
+        },
+        'datastore': {
+            'tab': 'MyOtherTasks',
+            'task': 'SmartState Analysis for [{}]',
+            'state': "Finished"
+        },
+        'cluster': {
+            'tab': 'MyOtherTasks',
+            'task': 'SmartState Analysis for [{}]',
+            'state': "Finished"
+        }
+    }[task_type]
+    return all_tasks_finished(
+        destination=tabs_data['tab'],
+        task_name=tabs_data['task'].format(name),
+        expected_status=tabs_data['state'],
+        expected_num_of_tasks=expected_num_of_tasks
+    )
+
+
+def all_tasks_finished(destination, task_name, expected_status, expected_num_of_tasks):
+    """ Check if all tasks with same task name states are finished - if not, reload page"""
+    view = navigate_to(Tasks, destination)
+    tab_view = getattr(view.tabs, destination.lower())
+
+    # task_name change from str to support also regular expression pattern
+    task_name = re.compile(task_name)
+    # expected_status change from str to support also regular expression pattern
+    expected_status = re.compile(expected_status, re.IGNORECASE)
+
+    try:
+        rows = list(tab_view.table.rows(task_name=task_name, state=expected_status))
+    except IndexError:
+        logger.warn('IndexError exception suppressed when searching for task row, no match found.')
+        return False
+
+    # check state = finished for all tasks
+    return expected_num_of_tasks == len(rows), len(rows)
+
+
+def are_all_analysis_finished(name, expected_num_of_tasks, task_type='container',
+                              silent_failure=False, clear_tasks_after_success=False):
+    """ Check if all tasks analysis states are finished with no errors"""
+
+    tabs_data = {
+        'container': {
+            'tab': 'AllTasks',
+            'task': '{}',
+            'state': 'finished'
+        },
+        'vm': {
+            'tab': 'AllTasks',
+            'task': 'Scan from Vm {}',
+            'state': 'finished'
+        },
+        'host': {
+            'tab': 'MyOtherTasks',
+            'task': "SmartState Analysis for '{}'",
+            'state': 'Finished'
+        },
+        'datastore': {
+            'tab': 'MyOtherTasks',
+            'task': 'SmartState Analysis for [{}]',
+            'state': "Finished"
+        },
+        'cluster': {
+            'tab': 'MyOtherTasks',
+            'task': 'SmartState Analysis for [{}]',
+            'state': "Finished"}
+    }[task_type]
+    return all_analysis_finished(destination=tabs_data['tab'],
+                            task_name=tabs_data['task'].format(name),
+                            expected_status=tabs_data['state'],
+                            expected_num_of_tasks=expected_num_of_tasks,
+                            silent_failure=silent_failure,
+                            clear_tasks_after_success=clear_tasks_after_success)
+
+
+def all_analysis_finished(destination, task_name, expected_status, expected_num_of_tasks,
```

Comment:
Same as above - the method name doesn't really match the method behavior.

Consider wrapping the existing methods for `is_task_finished` and `is_analysis_finished` and call them for each task of the given name.

Reply:
Rename “are_all_analysis_finished” to  “are_all_tasks_analysis_pass”
Rename “all_analysis_finished” to  “all_tasks_analysis_pass”

I couldn’t  use “is_task_finished” for multiple task, since tasks table have no unique key.
We use the task name as a filter on this task table, but in our case there could be some container images with same name , so task name that includes Container Image name is not unique key that I could use.

In order to avoid duplication as you suggest , I use “all_tasks_analysis_pass”, that can be use  also for single task.
Therefore I delete “is_task_finished” (internal method have no usage beside in tasks module).
In addition I change “is_analysis_finished” to call “all_tasks_analysis_pass”.